core/primitives/src/sharding.rs

```diff
@@ -56,14 +56,29 @@ impl From<CryptoHash> for ChunkHash {
     }
 }
 
-#[derive(Debug, PartialEq, BorshSerialize, BorshDeserialize)]
+#[derive(Clone, Debug, PartialEq, BorshSerialize, BorshDeserialize)]
 pub struct ShardInfo(pub ShardId, pub ChunkHash);
 
 /// Contains the information that is used to sync state for shards as epochs switch
-#[derive(Debug, PartialEq, BorshSerialize, BorshDeserialize)]
+#[derive(Clone, Debug, PartialEq, BorshSerialize, BorshDeserialize)]
 pub struct StateSyncInfo {
-    /// The first block of the epoch for which syncing is happening
-    pub epoch_tail_hash: CryptoHash,
+    /// For now this is always set to 0. This is included because an improvement we might want to make in the future
+    /// is that when syncing to the current epoch's state, we currently wait for two new chunks in each shard, but
+    /// with some changes to the meaning of the "sync_hash", we should only need to wait for one. So this is included
+    /// in order to allow for this change in the future without needing another database migration.
+    pub state_sync_version: u32,
```

Comment:
Yeah I think that’s prob better, just in case there’s something about the struct we want to change without needing a db upgrade. Updated, and also tested again with https://github.com/marcelo-gonzalez/nearcore/blob/catchup-migration-test/pytest/tests/sanity/state_sync_info_migration.py